File: MošPosudit.Services/DataBase/Data/RentalItem.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace MošPosudit.Services.DataBase.Data
{
    public class RentalItem
    {
        [Key]
        public int Id { get; set; }

        [Required]
        public int RentalId { get; set; }

        [Required]
        public int ToolId { get; set; }

        [Required]
        [Range(1, int.MaxValue)]
        public int Quantity { get; set; }

        [Required]
        [Range(0.01, double.MaxValue)]
        public decimal DailyRate { get; set; }

        public string? Notes { get; set; }

        // Navigation properties
        [ForeignKey("RentalId")]
        public Rental Rental { get; set; }
        [ForeignKey("ToolId")]
        public Tool Tool { get; set; }
    }
} 

File: MosPosudit.UI/shared/lib/models/rental_item.dart
class RentalItemModel {
  final int id;
  final int rentalId;
  final int toolId;
  final int quantity;
  final num dailyRate;
  final String? notes;

  RentalItemModel({
    required this.id,
    required this.rentalId,
    required this.toolId,
    required this.quantity,
    required this.dailyRate,
    this.notes,
  });

  factory RentalItemModel.fromJson(Map<String, dynamic> json) => RentalItemModel(
        id: json['id'],
        rentalId: json['rentalId'],
        toolId: json['toolId'],
        quantity: json['quantity'],
        dailyRate: json['dailyRate'],
        notes: json['notes'],
      );

  Map<String, dynamic> toJson() => {
        'id': id,
        'rentalId': rentalId,
        'toolId': toolId,
        'quantity': quantity,
        'dailyRate': dailyRate,
        'notes': notes,
      };
}

